add more diagrams + jump to quickstart

andrewlawhh opened the pull request.

chester-leung commented on federated-xgboost/README.md:
```diff
@@ -3,10 +3,29 @@ Federated learning allows multiple parties to collaboratively learn a shared mod
 
 In the federated setting, a central party has a basic model that is initially broadcast to all parties. Each party locally trains the model with its own data, then sends a summary of the updates to the model back to the central party. In the decision tree case, parties would be sending the best local feature splits back to the central party. The central party then aggregates all updates, updates  ...
```
Can you add a note here to look at the diagrams you've added for more information about splits and aggregation?

chester-leung commented on federated-xgboost/README.md:
```diff
@@ -3,10 +3,29 @@ Federated learning allows multiple parties to collaboratively learn a shared mod
+![diagram 3](./images/detail_3.png)  
+
+![diagram 4](./images/detail_4.png)  
+
```
Can you enumerate the steps to show that one comes after another? Also is there any way to do two diagrams in a row? So step 1 and step 2 would be in one row, and step 3 and 4 underneath that? I think it'd look better in the README. Thanks!

chester-leung commented on federated-xgboost/README.md:
```diff
@@ -59,6 +78,4 @@ The following flags must be specified when running the `start_job.sh`
     
 ### Notes
-* This has only been tested with Python 3
-* `FederatedXGBoost.py` is a wrapper that simplifies the data loading, training, and evaluation process. 
```
Any reason you removed these two bullets?

andrewlawhh commented:
Added numbers and spacing, put the notes back (I deleted on accident mb)
Couldn't figure out how to inline the pictures though..

And I know it's not the prettiest so lmk if its still confusing or w/e

chester-leung commented on federated-xgboost/README.md:
```diff
@@ -1,12 +1,33 @@
 
+[QUICKSTART](#quickstart)
+
+### _High Level_
```
Could you remove the italics here? I think having the h3 header should be enough.

chester-leung commented on federated-xgboost/README.md:
```diff
@@ -1,12 +1,33 @@
+
+![federated diagram](./images/federated-xgboost-diagram.png)
+
+### _Finding a Split_
```
Here too.

chester-leung commented on federated-xgboost/README.md:
```diff
@@ -1,12 +1,33 @@
 
+[QUICKSTART](#quickstart)
+
+### _High Level_
```
Can you remove the italics here and leave it as a h3 header? I think that should be enough

andrewlawhh replied:
done

chester-leung commented on federated-xgboost/README.md:
```diff
@@ -1,12 +1,33 @@
+
+![federated diagram](./images/federated-xgboost-diagram.png)
+
+### _Finding a Split_
```
Here too

andrewlawhh replied:
done

chester-leung commented on federated-xgboost/README.md:
```diff
@@ -1,12 +1,33 @@
+
+Stopping conditions include  
+ - reaching the max_depth configurable parameter in the training method
+ - when no split is found to give a benefit score greater than some specified value gamma
```
Can you change these `-` to `*`?  Circular bullets may look better. Please also make sure the three bullets are lined up -- I think there's an extra space in front of the `-` right now

andrewlawhh replied:
done

chester-leung commented on federated-xgboost/README.md:
```diff
@@ -1,12 +1,33 @@
+
+ ![diagram 3](./images/detail_3.png)
+ 
+ ![diagram 4](./images/detail_4.png)
```
Can you try left aligning the steps so that they're all in the same place? Not sure if that'll look better but the steps look a little strange right now. 

I also wonder whether the README is the best place to put this. Maybe we can have a link in the README to a different file that shows this diagram. What are your thoughts?

andrewlawhh replied:
Yeah, I think thats better than my awkward jump to Quickstart link thing

andrewlawhh commented on federated-xgboost/README.md:
```diff
@@ -61,4 +82,4 @@ The following flags must be specified when running the `start_job.sh`
 * This has only been tested with Python 3
 * `FederatedXGBoost.py` is a wrapper that simplifies the data loading, training, and evaluation process. 
-* The `--sync-dst-dir` option in the `dmlc-submit` command copies everything in the passed in directory to all worker machines. This means that the training script can initially only be on the tracker machine, and will be automatically copied over to all parties once the job is submitted. 
+* The --sync-dst-dir option in the dmlc-submit command copies everything in the passed in directory to all worker machines. This means that the training script can initially only be on the tracker machine, and will be automatically copied over to all parties once the job is submitted.
```
done

chester-leung commented:
Thank you!
